src/PrestaShopBundle/Resources/config/services/adapter/language.yml

```diff
@@ -73,3 +73,11 @@ services:
     tags:
       - name: tactician.handler
         command: 'PrestaShop\PrestaShop\Core\Domain\Language\Command\DeleteLanguageCommand'
+
+  PrestaShop\PrestaShop\Adapter\Language\ContextLanguageProvider:
+    autowire: true
+    public: false
+    arguments:
+      $context: '@prestashop.adapter.legacy.context'
+
+  PrestaShop\PrestaShop\Core\Language\ContextLanguageProviderInterface: '@PrestaShop\PrestaShop\Adapter\Language\ContextLanguageProvider'
```

Comment:
```suggestion
  PrestaShop\PrestaShop\Core\Language\ContextLanguageProviderInterface:
    class: PrestaShop\PrestaShop\Adapter\Language\ContextLanguageProvider
    autowire: true
    public: false
    arguments:
      $context: '@prestashop.adapter.legacy.context'
```

Reply:
I agree it's a bit cleaner, but it also feels a bit weird that a core service definition is defined in an adapter service yaml file 🤔

Shouldn't we have the adapter definition as @zuk3975 suggested here, and another core definition file that would contain `PrestaShop\PrestaShop\Core\Language\ContextLanguageProviderInterface: '@PrestaShop\PrestaShop\Adapter\Language\ContextLanguageProvider'` which indicates which adapter service actually implements the interface?

Anyway this is going to be modified when we define the context replacement architecture so I don't think it's too bad even if we accept as is since it's gonna be modified later anyway